lib/reek/smells/uncommunicative_module_name.rb

```diff
@@ -49,21 +49,35 @@ def self.contexts # :nodoc:
       #
       # @return [Array<SmellWarning>]
       #
-      # :reek:Duplication { allow_calls: [ to_s ] }
-      def examine_context(ctx)
-        reject_names = value(REJECT_KEY, ctx, DEFAULT_REJECT_SET)
-        accept_names = value(ACCEPT_KEY, ctx, DEFAULT_ACCEPT_SET)
-        exp = ctx.exp
-        name = exp.simple_name.to_s
-        return [] if accept_names.include?(ctx.full_name)
-        var = name.gsub(/^[@\*\&]*/, '')
-        return [] if accept_names.include?(var)
-        return [] unless reject_names.find { |patt| patt =~ var }
+      def examine_context(context)
+        fully_qualified_name = context.full_name
+        exp                  = context.exp
+        module_name          = exp.simple_name
+
+        return [] if acceptable_name?(context: context,
+                                      module_name: module_name,
+                                      fully_qualified_name: fully_qualified_name)
+
         [smell_warning(
-          context: ctx,
+          context: context,
           lines: [exp.line],
-          message: "has the name '#{name}'",
-          parameters: { name: name })]
+          message: "has the name '#{module_name}'",
+          parameters: { name: module_name })]
+      end
+
+      private
+
+      def acceptable_name?(context:, module_name:, fully_qualified_name:)
```

Comment:
Yep, thats what I wanted to avoid, having those raises everywhere. 

But you're right, we introduced this pattern and we should see it through, will change it.